tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ConstSimplifierBase.scala

```diff
@@ -29,49 +29,170 @@ abstract class ConstSimplifierBase {
       }
 
     case OperEx(TlaArithOper.minus, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
       ValEx(TlaInt(left - right))(intTag)
 
-    case ex @ OperEx(TlaArithOper.minus, NameEx(left), NameEx(right)) =>
-      if (left == right) ValEx(TlaInt(0))(intTag) else ex // this actually happens
+    // 0 - x = -x
+    case OperEx(TlaArithOper.minus, ValEx(TlaInt(left)), rightEx) =>
+      if (left == 0) {
+        OperEx(TlaArithOper.uminus, rightEx)(intTag)
+      } else {
+        OperEx(TlaArithOper.minus, ValEx(TlaInt(left))(intTag), rightEx)(intTag)
+      }
+
+    // x - 0 = x
+    case OperEx(TlaArithOper.minus, leftEx, ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        leftEx
+      } else {
+        OperEx(TlaArithOper.minus, leftEx, ValEx(TlaInt(right))(intTag))(intTag)
+      }
+
+    // x - x = 0
+    case ex @ OperEx(TlaArithOper.minus, leftEx, rightEx) =>
+      if (leftEx == rightEx) ValEx(TlaInt(0))(intTag) else ex // this actually happens
 
     case OperEx(TlaArithOper.mult, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
-      if (left == 0 || right == 0) {
-        ValEx(TlaInt(BigInt(0)))(intTag)
+      ValEx(TlaInt(left * right))(intTag)
+
+    // 0 * x = 0
+    // 1 * x = x
+    case OperEx(TlaArithOper.mult, ValEx(TlaInt(left)), rightEx) =>
+      if (left == 0) {
+        ValEx(TlaInt(0))(intTag)
       } else if (left == 1) {
-        ValEx(TlaInt(right))(intTag)
+        rightEx
+      } else {
+        OperEx(TlaArithOper.mult, ValEx(TlaInt(left))(intTag), rightEx)(intTag)
+      }
+
+    // x * 0 = 0
+    // x * 1 = x
+    case OperEx(TlaArithOper.mult, leftEx, ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        ValEx(TlaInt(0))(intTag)
+      } else if (right == 1) {
+        leftEx
+      } else {
+        OperEx(TlaArithOper.mult, leftEx, ValEx(TlaInt(right))(intTag))(intTag)
+      }
+
+    case ex @ OperEx(TlaArithOper.div, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        throw new IllegalArgumentException(s"Division by zero at ${ex.toString}")
+      } else {
+        ValEx(TlaInt(left / right))(intTag)
+      }
+
+    // 0 / x = 0
+    case OperEx(TlaArithOper.div, ValEx(TlaInt(left)), rightEx) =>
+      if (left == 0) {
+        ValEx(TlaInt(0))(intTag)
+      } else {
+        OperEx(TlaArithOper.div, ValEx(TlaInt(left))(intTag), rightEx)(intTag)
+      }
+
+    // x / 1 = x
+    case ex @ OperEx(TlaArithOper.div, leftEx, ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        throw new IllegalArgumentException(s"Division by zero at ${ex.toString}")
+      } else if (right == 1) {
+        leftEx
+      } else {
+        OperEx(TlaArithOper.div, leftEx, ValEx(TlaInt(right))(intTag))(intTag)
+      }
+
+    // x / x = 1
+    case ex @ OperEx(TlaArithOper.div, leftEx, rightEx) =>
+      if (leftEx == rightEx) {
+        ValEx(TlaInt(1))(intTag)
+      } else {
+        OperEx(TlaArithOper.div, leftEx, rightEx)(intTag)
```

Comment:
Could this be:

```suggestion
        ex
```

if not for some reason, then I think the `ex` is unused in this case?